src/LibDeclarations/webxr.nativeextensions.d.ts

```diff
@@ -1,11 +1,82 @@
-// This file contains native only extensions for WebXR  These APIs are not supported in the browser yet.
+// This file contains native only extensions for WebXR. These APIs are not supported in the browser yet.
 // They are intended for use with either Babylon Native https://github.com/BabylonJS/BabylonNative or
 // Babylon React Native: https://github.com/BabylonJS/BabylonReactNative
 
+// Open question: Do better types exist to use for vectors/quaternions?
+interface XRVector {
+    x: number;
+    y: number;
+    z: number;
+}
+
+interface XRQuaternion {
+    x: number;
+    y: number;
+    z: number;
+    w: number;
+}
+
+// Open question: Should XRGeometryType be defined or should all types be strings?
+type XRGeometryType = "unknown" | "background" | "wall" | "floor" | "ceiling" | "platform";
+
+interface XRFieldOfView {
+    angleLeft: number;
+    angleRight: number;
+    angleUp: number;
+    angleDown: number;
+}
+
+interface XRFrustum {
+    position: XRVector;
+    rotation: XRQuaternion;
+    fieldOfView: XRFieldOfView;
+    farDistance: number;
+}
+
+interface XRPlane {
+    // Open question: Should geometry id's and types be declared on XRPlanes or queried through other means?
+    geometryId?: number;
+    geometryType?: XRGeometryType | string;
+}
+
+// Open question: Should XRMesh have a concept of a XRReferenceSpace similar to XRPlane?
+interface XRMesh {
+    positions: Float32Array;
+    indices: Uint32Array;
+    normals?: Float32Array;
+    // Open question: Do we need lastChangedTime declared for XRMeshes?
+    lastChangedTime: number;
+    // Open question: Should geometry id's and types be declared on XRMeshes or queried through other means?
+    geometryId?: number;
+    geometryType?: XRGeometryType | string;
+}
+
+type XRDetectionBoundaryType = "frustum" | "sphere" | "box";
+
+interface XRDetectionBoundary {
+    isStationary?: boolean;
+    type?: XRDetectionBoundaryType | string;

```

Comment:
Why do we need the `| string` part here? I thought maybe for additional "unknown" types, but then I'm not sure how that would be useful because if it's unknown we wouldn't know what to do with it right?

Reply:
Good catch. I agree we don't need a string here. Having to implement the native backend for a specific detection boundary type would warrant updating the XRDetectionBoundaryType definition.